src/datadog/telemetry/telemetry.cpp

```diff
@@ -116,4 +116,14 @@ void report_error_log(std::string message) {
              instance());
 }
 
+void report_error_log(std::string message, std::string stacktrace) {
```

Comment:
If my other suggestions are accepted, here we could use:
```suggestion
void report_error_log(std::string message, std::optional<std::string> stacktrace) {
  std::visit(details::Overload{
                 [&](Telemetry& telemetry) { telemetry.log_error(message, stacktrace); },
                 [](auto&&) {},
             },
             instance());
}
```